content/docs/ref/index.md

```diff
@@ -1,4 +1,17 @@
-# Command Reference
+# CML Command Reference
+
+CML provides a number of commands to help package the outputs of ML workflows
+into a CML report. These may include numeric data or model performance
+visualizations. Let's look at the typical sequence (after
+[configuration](/doc/config)):
+
+∞ `cml runner` launches a runner hosted by a cloud compute provider or
+[on-premise](/doc/self-hosted-runners).
+
+∞ `cml pr` commits a set of files to a new branch and create a pull request.
+
+∞ `cml comment` posts a Markdown report as a comment on a commit or pull/merge
+request.
```

Comment:
```suggestion
```

Reply:
This is where I moved the list of selected commands from Usage, to have something similar to https://dvc.org/doc/command-reference#typical-dvc-workflow . Of course we don't need to have the same structure or strategies as DVC docs but this one also applies, I think? There's a typical workflow (to give readers an expectation of which commands they'll probably use most). We could update the list though, but I need help with that (almost never use CML atm).

Reply:
I strongly object to inserting synopsis/verbose explanations/justification/problem-context into "command reference". To me a "command ref" is meant to be:

- closer to a "glossary" or even "index page" in a book
  + super concise, dry, to-the-point
- aimed at power-users who know exactly what they conceptually need
  + e.g. can't quite remember the spelling/vocab (e.g. reader is wondering "was the flag called `--machine-type` or `--cloud-type`?")
- links to other more prosaic pages for fuller explanations (no inline explanations)
- includes common error codes/errors/edge cases/FAQ help

And I think this should apply to all software docs sites with zero exceptions.

Reply:
> a "cmd ref" is meant to be ... super concise, dry, to-the-point

It's up to us to decide. The approach I'm familiar with from DVC and now MLEM, GTO is that the terminal-based docs (`--help` output) is the dry/ purely operational one.

> ... aimed at power-users who know exactly what they conceptually need

In our current docs framework (see [Ref-type content](https://www.notion.so/iterative/Mid-Low-level-content-f435238a76af46feb91e0d0321416c79#7c0b737885c34a2ab06240273edb995d)) we go beyond this, into "self-contained UI manuals". Self-contained meaning it's a) comprehensive, b) explains (as brief as possible but still) enough to understand the key concepts, and c) links to other refs are enough to fill any gaps (bc/ things are explained there too).

> links to other more prosaic pages

Yep we also include those but in admons as to note this is for further reading/ deeper learning (optional).

> includes ...

The exact template for ref pages is def. flexible depending on product needs.

Reply:
> keep "cmd ref" as a concise replacement for CLI --help

To reiterate, this is up to you. May just want to involve @shcheklein or others and then make the call. If it differs from other products let's just write it somewhere so we remember in the future. For now feel free to commit this suggestion (up to you @casperdcl). Thanks

Reply:
p.s. useful guide we have tried to follow before (esp. [for DVC](https://github.com/iterative/dvc/issues/6405)): https://clig.dev/#documentation

> **Provide web-based documentation.** ... The web is the most inclusive documentation format available.
>
> **Provide terminal-based documentation.** ... it’s fast to access, it stays in sync with the specific installed version of the tool, and it works without an internet connection.

Reply:
Hey folks, my take on this. And I agree that it's better for us to apply same rules (or apply exception consciously for a reason that we understand).

Since here you've been discussing the index page, not actual command reference pages. I don't have a particularly strong opinion about it tbh. Practically, this page indeed can serve as an _index_ page:

```
closer to a "glossary" or even "index page" in a book
super concise, dry, to-the-point efficient
```

even its name implies this. 

So, it doesn't hurt to have a table with commands. Something similar to a cheat sheet. If project doesn't have UG it might make sense to have diagrams, etc that would map all existing commands into workflow, etc. 

I just don't see how that can hurt.

------------

If we talk about specific pages inside command reference (an actual command pages). A page should have enough information to understand the semantics of an operation and what to expect from it if I run it in different ways (e.g. `dvc gc` - it should be clear that it collects references, removes those that are not used. It should be clear that `-c` affects the local cache _and_ remote, etc, etc). The way to achieve that may be a bit different, but I would say from the most important to less important:

- options and their descriptions. In some case it's fine to repeat the same sentence or two from CLI, in a lot of cases that's not enough. We should be of course trying to optimize so that both are close as much as possible (means that ppl can read less, can visit web less, etc). I just don't believe it's realistic to rely on one-two sentences there.
- examples - any example, especially if it comes from a practical common problem worth a lot
- description - I would expect ppl to read this less. Arguably this can become a part of UG, or something else.

Command reference in my head maps to a man page, to pages like this - https://git-scm.com/docs/git-checkout

Reply:
> inapplicable to CML, which effectively does not have terminal-based docs

Good point. Let's keep that in mind but still the web format allows us more flexibility (no need to keep refs super dry IMO).